README.md
# PersonalAssistant
<a href="https://www.buymeacoffee.com/klingo" target="_blank"><img src="https://cdn.buymeacoffee.com/buttons/default-yellow.png" alt="Buy Me A Coffee"></a>

PersonalAssistant, an Add-on for '[The Elder Scrolls Online](https://www.elderscrollsonline.com/ "Home - The Elder Scrolls Online")'

***

## Download
You can always download the latest version here: http://www.esoui.com/downloads/info381-PersonalAssistant.html

***

## Roadmap

At the current time I do not have much time fo ESO, respectively my priorities lie elsewhere. Nonetheless, every now and then I find some time again to work on this addon.

On a higher level, this is the rough roadmap for PersonalAssistant:

| Release | Focus | Content                                                                                                     |
|---------|------------|-------------------------------------------------------------------------------------------------------------|
| 2.5.x   | PAGeneral  | - Refactoring profile management                                                                              |
| 2.6.x   | PABanking  | - Advanced transfer rules<br>- Additional FCOIS support<br>- Manual transfer button                               |
| 2.7.x   | PALoot     | - Track known recipes, traits and motifs across all characters<br>- Notify about group loots<br>- Auto-Laundering |
| 2.8.x   | PAWorker   | - Auto-Deconstruct Items                                                                                      |                                                                       |
 (subject to change)

***

## How to use
PersonalAssistant is an Add-on for '[The Elder Scrolls Online](https://www.elderscrollsonline.com/ "Home - The Elder Scrolls Online")' and consists of multiple nested Add-ons that can independently be turned on or off. The following section will explain in more detail how to use the different parts. 

#### PersonalAssistant
![alt text][pag-menu]

tbd

#### PersonalAssistant Banking
![alt text][pab-menu]

tbd

#### PersonalAssistant Integration
![alt text][pai-menu]

tbd

#### PersonalAssistant Junk
![alt text][paj-menu]

tbd

#### PersonalAssistant Loot
![alt text][pal-menu]

![alt text][pal-inventory]

tbd

#### PersonalAssistant Repair
![alt text][par-menu]

tbd




***

## Disclaimer

**Disclaimer:**
This Add-on is not created by, affiliated with or sponsored by ZeniMax Media Inc. or its affiliates. The Elder Scrolls® and related logos are registered trademarks or trademarks of ZeniMax Media Inc. in the United States and/or other countries. All rights reserved.


[pag-menu]: ./info/images/PAG.png "PersonalAssistant General Menu"
[pab-menu]: ./info/images/PAB.png "PersonalAssistant Banking Menu"
[pai-menu]: ./info/images/PAI.png "PersonalAssistant Integration Menu"
[paj-menu]: ./info/images/PAJ.png "PersonalAssistant Junk Menu"
[pal-menu]: ./info/images/PAL.png "PersonalAssistant Loot Menu"
[pal-inventory]: ./info/images/PAL_Inventory.png "PersonalAssistant Loot Inventory View"
[par-menu]: ./info/images/PAR.png "PersonalAssistant Repair Menu"

info/changelog.txt
2.5.24 (October 16, 2021)
- added support for Deadlands (APIVersion 101032) ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/408"]#408[/URL])
- updated minimum required version of dependencies ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/405"]#405[/URL])
- fixed a PABanking issue for items marked with FCOIS.Locked flag ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/406"]#406[/URL])
- removed a PAJunk check for "isUnique" when marking items as junk ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/407"]#407[/URL])

2.5.23 (August 15 2021)
- fixed an issue in the debugging code ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/399"]#399[/URL])

2.5.22 (August 14, 2021)
- updated PAJunk's item quality dropdowns with legendary quality ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/397"]#397[/URL])
- fixed a PAJunk issue where the keybind buttons were not correctly hidden ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/396"]#396[/URL])
- fixed PARepair's use of Crown Repair Kits  ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/394"]#394[/URL])
- removed some PABanking caching logic that could sometimes cause problems ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/395"]#395[/URL])

2.5.21 (August 03, 2021)
- disabled PARepair's use of Crown Repair Kits until they can be handled properly ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/393"]#393[/URL])

2.5.20 (July 31, 2021)
- fixed a PALoot issue where the item icons were not updated after PABanking did run ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/392"]#392[/URL])

2.5.19 (July 28, 2021)
- added support for Waking Flame (APIVersion 101031) ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/389"]#389[/URL])
- fixed a PABanking issue where the "Run PABanking" button was not correctly named ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/390"]#390[/URL])

2.5.18 (July 25, 2021)
- added a keybind shortcut for traveling to the primary house ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/379"]#379[/URL])
- added new PABanking settings for additional FCOIS marks ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/378"]#378[/URL])
- added new PARepair setting to also use Crown Repair Kits ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/380"]#380[/URL])
- added new PALoot icons for companion items  ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/383"]#383[/URL] / [URL="https://github.com/klingo/ESO-PersonalAssistant/pull/386"]#386[/URL])
- fixed a PABanking issue with crafting items that have active custom rules ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/375"]#375[/URL])
- fixed a PALoot issue with looting and binding uncollected set items ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/381"]#381[/URL] / [URL="https://github.com/klingo/ESO-PersonalAssistant/pull/384"]#384[/URL])
- fixed a PAJunk issue where companion items were wrongly shown as 'junkable' ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/387"]#387[/URL])
- fixed potential issues with uninitialized profile counter and undefined bag size ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/374"]#374[/URL] / [URL="https://github.com/klingo/ESO-PersonalAssistant/pull/376"]#376[/URL])

2.5.17 (May 09, 2021)
- updated PABanking to avoid reversing item transfers in case of active custom rules ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/369"]#369[/URL])
- fixed an issue with PAJunk not being loaded properly in combination with PAIntegration ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/368"]#368[/URL])

2.5.16 (May 09, 2021)
- added support for Blackwood (APIVersion 100035) ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/367"]#367[/URL])
- updated Russian translations from Nesferatum to 2.5.15 ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/360"]#360[/URL])
- fixed (hopefully) some issues with FCOIS and missing active profiles ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/365"]#365[/URL] / [URL="https://github.com/klingo/ESO-PersonalAssistant/pull/366"]#366[/URL])

2.5.15 (April 25, 2021)
- updated Russian translations from Nesferatum to 2.5.14 ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/358"]#358[/URL])
- removed profile selection for "PersonalAssistant", and other minor improvements ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/359"]#359[/URL])

2.5.14 (April 24, 2021)
- fixed various additional issues found in version 2.5.11/12/13 ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/357"]#357[/URL])

2.5.13 (April 24, 2021)
- fixed various smaller issues found in version 2.5.11/12 ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/356"]#356[/URL])

2.5.12 (April 24, 2021)
- fixed an issue where the crafting item section of PABanking did not work anymore ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/355"]#355[/URL])

2.5.11 (April 23, 2021)
- added new PAJunk quest protection for thieves guild daily 'The Covetous Countess' ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/337"]#337[/URL])
- added new PALoot setting to inform you about loot that is not yet in your set collection ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/350"]#350[/URL])
- updated minimum required version of dependencies ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/338"]#338[/URL])
- updated profile management: every module has now it's own set of profiles ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/341"]#341[/URL])
- updated PABanking's trophies section into multiple ones for treasure maps, fragments and survey reports ([URL="https://github.com/klingo/ESO-PersonalAssistant/pull/336"]#336[/URL])

vars/Globals.lua
if PA_SavedVars						== nil then PA_SavedVars					= {} end
if PA_SavedVars.General				== nil then PA_SavedVars.General			= {} end
if PA_SavedVars.Profiles			== nil then PA_SavedVars.Profiles			= {} end
if PA_SavedVars.Repair				== nil then PA_SavedVars.Repair				= {} end
if PA_SavedVars.Banking				== nil then PA_SavedVars.Banking 			= {} end
if PA_SavedVars.Banking.ItemTypes 	== nil then PA_SavedVars.Banking.ItemTypes	= {} end
if PA_SavedVars.Loot				== nil then PA_SavedVars.Loot 			    = {} end
if PA_SavedVars.Junk				== nil then PA_SavedVars.Junk 			    = {} end

-- PersonalAssistant Constants
-- PA Banking
PAC_ITEMTYPE_IGNORE = 0
PAC_ITEMTYPE_DEPOSIT = 1
PAC_ITEMTYPE_WITHDRAWAL = 2
PAC_ITEMTYPE_INHERIT = 3
-- PA Loot
PAC_ITEMTYPE_LOOT = 1

PAC_OPERATOR_NONE = 0
PAC_OPERATOR_EQUAL = 1
-- PAC_OPERATOR_LESSTHAN = 2
PAC_OPERATOR_LESSTAHNEQAL = 3
-- PAC_OPERATOR_GREATERTHAN = 4
PAC_OPERATOR_GREATERTHANEQUAL = 5

-- PersonalAssistant General
PAG_MAX_PROFILES = 5

-- PersonalAssistant Banking
PAB_DEPOSIT_MAX_LOOPS = 3

PAB_STACKING_FULL = 0		-- 0: Full depositing/withdrawl
PAB_STACKING_CONTINUE = 1	-- 1: Continue existing stacks
PAB_STACKING_INCOMPLETE = 2	-- 2: Complete existing stacks

-- PersonalAssistant Colors
PAC_COL_WHITE = "|cFFFFFF"
PAC_COL_YELLOW = "|cFFFF00"
PAC_COL_LIGHT_BLUE = "|cB0B0FF"

-- =====================================================================================================================
-- = PA Banking
-- =================================
PAItemTypes = {}
-- PAItemTypes[index] = ItemType   ["" --> disabled itemType]
-- Alchemy
table.insert(PAItemTypes, ITEMTYPE_REAGENT) -- 00
table.insert(PAItemTypes, ITEMTYPE_POISON_BASE) -- 01
table.insert(PAItemTypes, ITEMTYPE_POTION_BASE) -- 02
-- Blacksmithing
table.insert(PAItemTypes, ITEMTYPE_BLACKSMITHING_RAW_MATERIAL) -- 03
table.insert(PAItemTypes, ITEMTYPE_BLACKSMITHING_MATERIAL) -- 04
table.insert(PAItemTypes, ITEMTYPE_BLACKSMITHING_BOOSTER) -- 05
-- Clothing
table.insert(PAItemTypes, ITEMTYPE_CLOTHIER_RAW_MATERIAL) -- 06
table.insert(PAItemTypes, ITEMTYPE_CLOTHIER_MATERIAL) -- 07
table.insert(PAItemTypes, ITEMTYPE_CLOTHIER_BOOSTER) -- 08
-- Woodworking
table.insert(PAItemTypes, ITEMTYPE_WOODWORKING_RAW_MATERIAL) -- 09
table.insert(PAItemTypes, ITEMTYPE_WOODWORKING_MATERIAL) -- 10
table.insert(PAItemTypes, ITEMTYPE_WOODWORKING_BOOSTER) -- 11
-- Jewelcrafting
table.insert(PAItemTypes, ITEMTYPE_JEWELRYCRAFTING_RAW_MATERIAL) -- 63
table.insert(PAItemTypes, ITEMTYPE_JEWELRYCRAFTING_MATERIAL) -- 64
table.insert(PAItemTypes, ITEMTYPE_JEWELRYCRAFTING_BOOSTER) -- 65
-- Enchanting
table.insert(PAItemTypes, ITEMTYPE_ENCHANTING_RUNE_ASPECT) -- 12
table.insert(PAItemTypes, ITEMTYPE_ENCHANTING_RUNE_ESSENCE) -- 13
table.insert(PAItemTypes, ITEMTYPE_ENCHANTING_RUNE_POTENCY) -- 14
table.insert(PAItemTypes, ITEMTYPE_GLYPH_ARMOR) -- 15
table.insert(PAItemTypes, ITEMTYPE_GLYPH_JEWELRY) -- 16
table.insert(PAItemTypes, ITEMTYPE_GLYPH_WEAPON) -- 17
-- Provisioning
table.insert(PAItemTypes, ITEMTYPE_INGREDIENT) -- 18
table.insert(PAItemTypes, ITEMTYPE_RECIPE) -- 19
-- Furnishing
table.insert(PAItemTypes, ITEMTYPE_FURNISHING_MATERIAL) -- 62
-- Others
table.insert(PAItemTypes, ITEMTYPE_DRINK) -- 20
table.insert(PAItemTypes, ITEMTYPE_FOOD) -- 21
table.insert(PAItemTypes, ITEMTYPE_POTION) -- 22
table.insert(PAItemTypes, ITEMTYPE_POISON) -- NEW!!!
table.insert(PAItemTypes, ITEMTYPE_ARMOR_TRAIT) -- 23
table.insert(PAItemTypes, ITEMTYPE_WEAPON_TRAIT) -- 24
table.insert(PAItemTypes, ITEMTYPE_STYLE_MATERIAL) -- 25
table.insert(PAItemTypes, ITEMTYPE_RAW_MATERIAL) -- 26
table.insert(PAItemTypes, ITEMTYPE_FISH) -- 27

-- PersonalAssistant advanced ItemTypes
PAItemTypesAdvanced = {}
PAItemTypesAdvanced[0] = 30357	-- Lockpick
-- Can't use ITEMTYPE_LOCKPICK or SPECIALIZED_ITEMTYPE_LOCKPICK since the actual lockpicks are categorized as ITEMTYPE_TOOL (as of APIVersion 100018)

-- =====================================================================================================================
-- = PA Loot
-- =================================
-- PersonalAssistant Loot ItemTypes
PALoItemTypes = {}
-- Alchemy
table.insert(PALoItemTypes, ITEMTYPE_REAGENT)
table.insert(PALoItemTypes, ITEMTYPE_POTION_BASE)
table.insert(PALoItemTypes, ITEMTYPE_FURNISHING_MATERIAL)
-- Blacksmithing
table.insert(PALoItemTypes, ITEMTYPE_BLACKSMITHING_RAW_MATERIAL)
-- Clothing
table.insert(PALoItemTypes, ITEMTYPE_CLOTHIER_RAW_MATERIAL)
-- Woodworking
table.insert(PALoItemTypes, ITEMTYPE_WOODWORKING_RAW_MATERIAL)
-- Jewelcrafting
table.insert(PALoItemTypes, ITEMTYPE_JEWELRYCRAFTING_RAW_MATERIAL)
-- Enchanting
table.insert(PALoItemTypes, ITEMTYPE_ENCHANTING_RUNE_ASPECT)
table.insert(PALoItemTypes, ITEMTYPE_ENCHANTING_RUNE_ESSENCE)
table.insert(PALoItemTypes, ITEMTYPE_ENCHANTING_RUNE_POTENCY)
-- Provisioning
table.insert(PALoItemTypes, ITEMTYPE_INGREDIENT)
-- Fishing
table.insert(PALoItemTypes, ITEMTYPE_FISH)
table.insert(PALoItemTypes, ITEMTYPE_LURE)


-- =====================================================================================================================
-- = PA Junk
-- =================================
-- PersonalAssistant Junk ItemTypes
PAJItemTypes = {}